File: src/io/ivy/fawkes/beton/events/TestEvent.java

package io.ivy.fawkes.beton.events;

import pl.betoncraft.betonquest.core.QuestEvent;

import org.bukkit.event.Listener;

public class TestEvent extends QuestEvent implements Listener {
    public TestEvent(String playerID, String instructions) {
        super(playerID, instructions);
        System.out.println("This is a test.");
    }
}

File: src/ivy/fawkes/ext/beton/testevent.clj

(ns ivy.fawkes.ext.beton.testevent
  
  (:import [pl.betoncraft.betonquest.core QuestEvent]
           [org.bukkit Bukkit]
           [org.bukkit.event Listener])
  
  (:gen-class :name ivy.fawkes.ext.beton.TestEvent
              :init testevent
              :extends pl.betoncraft.betonquest.core.QuestEvent
              :implements [org.bukkit.event.Listener]))

(defn -testevent [player instructions]
  (println "Yes, ok. fine.")
  (.info (Bukkit/getLogger) "OMG YES IT WORKS"))
